deployment_scripts/appimage/build_appimage.sh

```diff
@@ -0,0 +1,225 @@
+
+  requirements_island="$ISLAND_PATH/requirements.txt"
+  # TODO: This is an ugly hack. PyInstaller is a build-time dependency and should
+  #		  not be installed as a runtime requirement.
+  sed '4d' $requirements_island | sponge $requirements_island
+
+  ${python_cmd} -m pip install -r "${requirements_island}"  --ignore-installed --prefix /usr --root=$APPDIR || handle_error
+}
+
+download_monkey_agent_binaries() {
+log_message "Downloading monkey agent binaries to ${ISLAND_BINARIES_PATH}"
+  mkdir -p "${ISLAND_BINARIES_PATH}" || handle_error
+  curl -L -o ${ISLAND_BINARIES_PATH}/${LINUX_32_BINARY_NAME} ${LINUX_32_BINARY_URL}
+  curl -L -o ${ISLAND_BINARIES_PATH}/${LINUX_64_BINARY_NAME} ${LINUX_64_BINARY_URL}
+  curl -L -o ${ISLAND_BINARIES_PATH}/${WINDOWS_32_BINARY_NAME} ${WINDOWS_32_BINARY_URL}
+  curl -L -o ${ISLAND_BINARIES_PATH}/${WINDOWS_64_BINARY_NAME} ${WINDOWS_64_BINARY_URL}
+
+  # Allow them to be executed
+  chmod a+x "$ISLAND_BINARIES_PATH/$LINUX_32_BINARY_NAME"
+  chmod a+x "$ISLAND_BINARIES_PATH/$LINUX_64_BINARY_NAME"
+}
+
+install_mongodb() {
+  log_message "Installing MongoDB"
+
+  mkdir -p $MONGO_PATH
+  "${ISLAND_PATH}"/linux/install_mongo.sh ${MONGO_PATH} || handle_error
+}
+
+generate_ssl_cert() {
+  log_message "Generating certificate"
+
+  chmod u+x "${ISLAND_PATH}"/linux/create_certificate.sh
+  "${ISLAND_PATH}"/linux/create_certificate.sh ${ISLAND_PATH}/cc
+}
+
+build_frontend() {
+    pushd "$ISLAND_PATH/cc/ui" || handle_error
+    npm install sass-loader node-sass webpack --save-dev
+    npm update
+
+    log_message "Generating front end"
+    npm run dist
+    popd || handle_error
+}
+
+build_appimage() {
+	log_message "Building AppImage"
+	appimage-builder --recipe monkey_island_builder.yml --log DEBUG --skip-appimage
+
+	# There is a bug or unwanted behavior in appimage-builder that causes issues
+	# if 32-bit binaries are present in the appimage. To work around this, we:
+	#   1. Build the AppDir with appimage-builder and skip building the appimage
+	#   2. Add the 32-bit binaries to the AppDir
+	#   3. Build the AppImage with appimage-builder from the already-built AppDir
+	#
+	# Note that appimage-builder replaces the interpreter on the monkey agent binaries
+	# when building the AppDir. This is unwanted as the monkey agents may execute in 
+	# environments where the AppImage isn't loaded.
+	#
+	# See https://github.com/AppImageCrafters/appimage-builder/issues/93 for more info.
+	download_monkey_agent_binaries
+
+	appimage-builder --recipe monkey_island_builder.yml --log DEBUG --skip-build
+}
+
+if is_root; then
+  log_message "Please don't run this script as root"
+  exit 1
+fi
+
+if ! has_sudo; then
+  log_message "You need root permissions for some of this script operations. \
+Run \`sudo -v\`, enter your password, and then re-run this script."
+  exit 1
+fi
+
+config_url="https://raw.githubusercontent.com/mssalvatore/monkey/linux-deploy-binaries/deployment_scripts/config"
```

Comment:
Pretty much looks like config in deployment script. You use the url because script hasn't downloaded the repo yet? My questions then: 1. why don't use a file from our repo? 2. If user is using this script without the whole repo, how did he get this script? Pulled only a specific folder? If we had usage docs, would be easier to understand

Reply:
I'll add a readme. I pretty much copied and reworked the linux deployment script.